integration_test.go

```diff
@@ -0,0 +1,475 @@
+package viamorbslam3_test
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+	slamConfig "go.viam.com/slam/config"
+	dataprocess "go.viam.com/slam/dataprocess"
+	slamTesthelper "go.viam.com/slam/testhelper"
+	"go.viam.com/test"
+	"go.viam.com/utils"
+	"go.viam.com/utils/artifact"
+
+	viamorbslam3 "github.com/viamrobotics/viam-orb-slam3"
+	"github.com/viamrobotics/viam-orb-slam3/internal/testhelper"
+)
+
+const (
+	dataInsertionMaxTimeoutMin = 3
+	orbSleepMs                 = 100
+)
+
+// Creates the vocabulary file required by the orbslam binary.
+func createVocabularyFile(name string) error {
+	source, err := os.Open(artifact.MustPath("slam/ORBvoc.txt"))
+	if err != nil {
+		return err
+	}
+	defer source.Close()
+	destination, err := os.Create(name + "/config/ORBvoc.txt")
+	if err != nil {
+		return err
+	}
+	defer destination.Close()
+	_, err = io.Copy(destination, source)
+	return err
+}
+
+// Releases an image or image pair to be served by the mock camera(s). If a pair of images is
+// released, it is released under a mutex, so that the images will be consumed in the same call
+// to getSimultaneousColorAndDepth().
```

Comment:
```suggestion
// releaseImages releases an image or image pair to be served by the mock camera(s). If a pair of images is
// released, it is released under a mutex so that the images will be consumed in the same call
// to getSimultaneousColorAndDepth().
```